mmhuman3d/data/data_structures/smc_reader.py

```diff
@@ -519,3 +533,153 @@ def get_depth_floor(self, device_id: int) -> dict:
             return device_dict['floor']
         else:
             raise KeyError(f'Kinect {device_id} has no floor data.')
+
+    def _get_keypoints2d(self, device, device_id, frame_id):
+        """Get keypoints2d projected from keypoints3d.
+
+        Args:
+            device (str):
+                Device name, should be Kinect or iPhone.
+            device_id (int):
+                ID of a device, starts from 0.
+            frame_id (int, list or None, optional):
+                int: frame id of one selected frame
+                list: a list of frame id
+                None: all frames will be returned
+                Defaults to None.
+
+        Returns:
+            Tuple[np.ndarray, np.ndarray]: keypoints2d and its mask
+        """
+        assert device in {'Kinect', 'iPhone'}
+        kps2d_dict = self.smc['Keypoints2D'][device][str(device_id)]
+        keypoints2d = kps2d_dict['keypoints2d'][...]
+        keypoints2d_mask = kps2d_dict['keypoints2d_mask'][...]
+
+        if frame_id is not None:
+            if isinstance(frame_id, int):
+                frame_id = [frame_id]
+            keypoints2d = keypoints2d[frame_id, ...]
+        return keypoints2d, keypoints2d_mask
```

Comment:
What is `keypoints2d_mask`? Is it from `convert_kps()`?

Reply:
`keypoints2d` and `keypoints2d_mask` are generated from mocap processing pipeline and stored in [human data format](https://github.com/open-mmlab/mmhuman3d/blob/main/docs/human_data.md#the-keys-and-values-supported-by-humandata-are-described-as-below). Then it is converted to a specific convention by `convert_kps()`.